### DISCUSSION TITLE
How to create a container of data?

### DISCUSSION THREAD

**Discussion by Linya-IronMan** [ramda/ramda#3312] [Q&A]
There is  a  concept  of functor in functional programing. The functor will take `of` and `map` api.
```javascript
const Container = function(x) {
  this.__value = x;
}

Container.of = function(x) {return new Container(x)};
Container.prototype.map = function(func) {return Container.of(func(this.__value))};
```
It's like this! 
Is there any function I can use in ramda.js?

**Comment by semmel** [comment]
Looks like the Identity monad. There plenty of implementations e.g. [crocks/Identity](https://crocks.dev/docs/crocks/Identity.html).

Ramda provides just the functions to work with data, not the implementations of such lawful data types.
